### ISSUE TITLE
Issues with cattle-nodes-AGENT

### ISSUE THREAD

**Issue by reymonlu** [rancher/rancher#16338]
**What kind of request is this (question/bug/enhancement/feature request):**
Question about cattle agent elements.

**Result:**
cattle agents pods running without crashing.

**Other details that may be helpful:**
I followed the steps of the installation (https://rancher.com/docs/rancher/v2.x/en/installation/). When it comes to install rancher with helm, I didn't know what to put in hostname. So i put a random hostname and add it to my hosts file to access to it. So I can access to my cluster through this hosname but when i look to all the pods that rancher install, I have this 
```
NAMESPACE       NAME                                      READY   STATUS             RESTARTS   AGE
cattle-system   cattle-cluster-agent-79449d74df-wgd9n     0/1     CrashLoopBackOff   29         2h
cattle-system   cattle-node-agent-6tt94                   0/1     CrashLoopBackOff   76         3d
cattle-system   cattle-node-agent-d725c                   0/1     CrashLoopBackOff   9          26m
cattle-system   cattle-node-agent-gkvgn                   0/1     CrashLoopBackOff   76         3d
```
I checked logs about those pods and that is the content:
```
INFO: Environment: CATTLE_ADDRESS=10.0.3.15 CATTLE_AGENT_CONNECT=true CATTLE_CA_CHECKSUM=3629a8263825a488cd5183af5fd500ad35bfcac7e7bdfe3a399c4e1e57a6e0d2 CATTLE_CLUSTER=false CATTLE_INTERNAL_ADDRESS= CATTLE_K8S_MANAGED=true CATTLE_NODE_NAME=slave1 CATTLE_SERVER=https://rancher.local.fr
ERROR: https://xxxxxx/ping is not accessible (Could not resolve host: xxxxxxx)
```
I would like to know how to resolve this problem and what is the purpose of these podes.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

**Cluster information**
- Cluster type  : Hosted
- Machine type : 3 VMS on debian strech (1 Master and 2 Workers)
- Kubernetes version : 1.2.1
- Rancher : 2.1.1

```
Server Version: 17.03.2-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-8-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 3.863 GiB
Name: master1
ID: 3525:RQVQ:JEZJ:6FIL:N7JP:IRYZ:DQ7G:WE6O:7Z53:V3HZ:7PCD:AROZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 192.168.56.102:5000
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support

```

**Comment by superseb** [comment]
The hostname is the DNS name where Rancher can be accessed. This is the DNS name to visit the UI/API, and where the node agents connect to when deployed in the cluster. If this is not a DNS name, it cannot be resolved. Putting aliases in `/etc/hosts` might work partly but the cluster agent pod is using the cluster DNS which resolves to upstream. So using proper DNS is the recommended approach. The purpose of the cattle-node-agents is communication with the nodes, the cattle-cluster-agent provides access to the cluster API.

Let me know if you think there is a bug here or if you have any follow-up questions.

**Comment by reymonlu** [comment]
Thanks for you fast answer. So in order to use kube-DNS I deleted rancher using helm delete --purge rancher. I deleted manually every elements in cattle-system namespace. After that, I install again rancher with helm with the hostname rancher which is the same that the service on cattle-system. Surprisingly, there are no more cattle-node-agents and cattle-cluster-agent and I had 3 "vociferous-crocodile-rancher" pods in my default namespace which use the same image that the rancher pods (rancher/rancher:v2.1.1) in the cattle-system namespace.
I would like to know why Cattle-agent pods disappeared ?
(sorry for my bad english, I try to be understable)
